### PR TITLE
fix: fix trigger retryStrategy

### PR DESCRIPTION
**Pull request by ThuWangzw**
Checklist:

* [] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

Fix the issue [#2928](https://github.com/argoproj/argo-events/issues/2928) that the trigger wasn't correctly retried as defined in the retryStrategy.

### CODE REVIEW

File: sensors/listener.go [hunk 1]
```diff
@@ -211,7 +211,16 @@ func (sensorCtx *SensorContext) listenEvents(ctx context.Context) error {
 			}
 
 			actionFunc := func(events map[string]cloudevents.Event) {
-				sensorCtx.triggerActions(ctx, sensor, events, trigger)
+				retryStrategy := trigger.RetryStrategy
+				if retryStrategy == nil {
+					retryStrategy = &apicommon.Backoff{Steps: 1}
```
Comment (terrytangyuan) [inline sensors/listener.go:216]: We should probably make this configurable

File: sensors/listener.go [hunk 2]
```diff
@@ -345,13 +354,21 @@ func (sensorCtx *SensorContext) triggerActions(ctx context.Context, sensor *v1al
...
-		sensorCtx.triggerWithRateLimit(ctx, sensor, trigger, eventsMapping, depNames, eventIDs)
+		return sensorCtx.triggerWithRateLimit(ctx, sensor, trigger, eventsMapping, depNames, eventIDs)
 	} else {
-		go sensorCtx.triggerWithRateLimit(ctx, sensor, trigger, eventsMapping, depNames, eventIDs)
+		go func() {
+			err := sensorCtx.triggerWithRateLimit(ctx, sensor, trigger, eventsMapping, depNames, eventIDs)
+			if err != nil {
+				// Log the error, and let it continue
+				logger := logging.FromContext(ctx)
+				logger.Warnw("Failed to execute a trigger", zap.Error(err), zap.String(logging.LabelTriggerName, trigger.Template.Name))
```
Comment (terrytangyuan) [inline sensors/listener.go:364]: logger.error instead of warn?

File: sensors/listener.go [hunk 3]
```diff
@@ -362,8 +379,10 @@ func (sensorCtx *SensorContext) triggerWithRateLimit(ctx context.Context, sensor
 		log.Errorw("Failed to execute a trigger", zap.Error(err), zap.String(logging.LabelTriggerName, trigger.Template.Name),
 			zap.Any("triggeredBy", depNames), zap.Any("triggeredByEvents", eventIDs))
 		sensorCtx.metrics.ActionFailed(sensor.Name, trigger.Template.Name)
+		return err
 	} else {
 		sensorCtx.metrics.ActionTriggered(sensor.Name, trigger.Template.Name)
+		return nil
```
Comment (terrytangyuan) [inline sensors/listener.go:385]: We can remove the else block and unindent these lines

File: sensors/listener.go (hunk 1 above)
Comment (whynowy) [reply sensors/listener.go:216]: There is a default backoff retry strategy.

File: sensors/listener.go (hunk 2 above)
Comment (ThuWangzw) [reply sensors/listener.go]: fixed 😄

**Review by whynowy — APPROVED** [review]
Thanks!

**Pull request merged**